services/src/test/java/org/keycloak/theme/beans/MessageFormatterMethodTest.java

```diff
@@ -64,6 +67,20 @@ public void test() throws TemplateModelException {
 
         msg = (String) fmt.exec(Arrays.asList(new SimpleScalar("key"),new SimpleNumber(2)));
         Assert.assertEquals("foo 2 foobar bar", msg);
+
+        msg = (String) fmt.exec(Arrays.asList(new SimpleScalar("simpleKey"),new SimpleNumber(2.5)));
+        Assert.assertEquals("2.5", msg);
+
+        msg = (String) fmt.exec(Arrays.asList(new SimpleScalar("simpleKey"),new SimpleDate(new Date(1763721018000L), 2)));
+        Assert.assertEquals("11/21/25, 11:30 AM", msg);
+
+        MessageFormatterMethod germanFmt = new MessageFormatterMethod(Locale.GERMANY, properties);
+
+        msg = (String) germanFmt.exec(Arrays.asList(new SimpleScalar("simpleKey"),new SimpleNumber(2.5)));
+        Assert.assertEquals("2,5", msg);
+
+        msg = (String) germanFmt.exec(Arrays.asList(new SimpleScalar("simpleKey"),new SimpleDate(new Date(1763721018000L), 2)));
+        Assert.assertEquals("21.11.25, 11:30", msg);
```

Comment:
It seems that your test is dependent on the time zone of the server. See below for a suggestion to normalize the time zone for the duration of the test:

```suggestion
        TimeZone defaultTimeZone = TimeZone.getDefault();
        try {
            TimeZone.setDefault(TimeZone.getTimeZone("UTC"));

            msg = (String) fmt.exec(Arrays.asList(new SimpleScalar("simpleKey"), new SimpleDate(new Date(1763721018000L), 2)));
            Assert.assertEquals("11/21/25, 10:30 AM", msg);

            MessageFormatterMethod germanFmt = new MessageFormatterMethod(Locale.GERMANY, properties);

            msg = (String) germanFmt.exec(Arrays.asList(new SimpleScalar("simpleKey"), new SimpleNumber(2.5)));
            Assert.assertEquals("2,5", msg);

            msg = (String) germanFmt.exec(Arrays.asList(new SimpleScalar("simpleKey"), new SimpleDate(new Date(1763721018000L), 2)));
            Assert.assertEquals("21.11.25, 10:30", msg);
        } finally {
            TimeZone.setDefault(defaultTimeZone);
        }
```